crates/sui-storage/src/lock_service.rs

```diff
@@ -84,7 +84,45 @@ enum LockServiceQueries {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct LockInfo {
+pub struct ObjectLockInfo {
+    requested_object_ref_lock_details: Option<Option<LockDetails>>,
+    latest_object_ref: ObjectRef,
```

Comment:
What's this field for?

Reply:
current `ObjectRef` in the lock table for that asked object. When `requested_obj_ref` is the presently locked version of that object, than `latest_object_ref` would be `requested_obj_ref`. This is to give the information when the `requested_obj_ref` does not match what is being locked at the moment.

Reply:
I see. So `ObjectLockInfo` is a return type used to express the lock result. But do you always need the ref? I feel like it probably should be an enum with different kind of lock result.

Reply:
enum sounds good.

Reply:
updated the PR with enum @lxfind